doc/fileformat/nmodl.rst

```diff
@@ -109,3 +109,35 @@ Arbor-specific features
     POST_EVENT(t) {
        g = g + (0.1*t)
     }
+
+Nernst
+------
+Many mechanisms make use of the reversal potential of an ion (``eX`` for ion ``X``).
+A popular equation for determining the reversal potential during the simulation is
+the `Nernst equation <https://en.wikipedia.org/wiki/Nernst_equation>`_.
+Both Arbor and NEURON make use of ``nernst``. Arbor implements it as a mechanism and
+NEURON implements it as a built-in method. However, the conditions for using the
+``nernst`` equation to change the reversal potential of an ion differ between the
+two simulators.
+
+1. In Arbor, the reversal potential of an ion remains equal to its initial value (which
+has to be set by the user) over the entire course of the simulation, unless another
+mechanism which alters that reversal potential is explictly added to the model
+(see :ref:`cppcablecell-revpot` for details).
+
+2. In NEURON, there is a rule which is evaluated under the hood to determine whether or
+not the reversal potential of an ion remains constant or is calculated using ``nernst``.
+The rule is documented `here <https://neuron.yale.edu/neuron/static/new_doc/modelspec/programmatic/ions.html>`_
+and can be summarized as follows:
+
+  If the internal or external concentration of an ion is **written**, and its reversal
+  potential is **read but not written**, then the nernst equation is used **continuously**
+  during the simulation to update the reversal potential of the ion.
+  And if the internal or external concentration of an ion is **read**, and its reversal
+  potential is **read but not written**, then the nernst equation is used **once** at the
+  beginning of the simulation to caluclate the reversal potential of the ion, and then
+  remains constant.
+  Otherwise, the reversal potential is set by the user and remains constant.
+
```

Comment:
There is one unclarity in the quoted NEURON docs, which I would love to do away with:
As written, it's unclear whether the rule applies to the local NMODL file only or the global
view is considered. 

Example
>  Assume a mechanism `A` that reads `eX` and writes to `Xi` is added. By the rules above, `nernst` is used.
>  Now we add mechanism `B` that writes to `eX`.
I _assume_ this results in `nernst` not being used, but is that correct?

Also, where is this rule applied by NEURON? Is having one CV overlap between `A` and `B` enough to change global behaviours? Or just for the overlap? (Does it matter?)

Reply:
The global view is considered. So for a given ion, you need to look at all the mechanisms that use/write it and reason about the effect. In the above example, `A` reads the `eX` value that is written by `B` and if `B` reads `Xi`, it reads the value written by `A`. I will make this clearer. 
As for whether `A` and `B` need to be on the same section for it to apply; I think yes. The docs indicate that this behavior is evaluated per `section`. Though I would need to check to be certain.

Reply:
So, tweaking the decomposition of the morphology into sections in NEURON potentially changes the treatment
of a globally used ion species? Or is that just for the overlapping CVs?

Sorry for asking so many questions on NEURON, but I expect to have to deal with them for quite some time in the 
future.

Reply:
So in Neuron, you can only add mechanisms on sections, not segments (CVs). I will run some examples to see what exactly happens in Neuron and get back to this question.

Reply:
Okay so based on my experiments: 
NEURON's rules for using nernst for the calculation of the reversal potential of an ion as stated above in this PR are applied per section (similar to arbor which can control the reversal potential of an ion at the granularity of a region). Since mechanisms can only be inserted on a section (not part of it) and since a NEURON segment (CV) can only belong to one section, there is no possibility for segment (CV) overlap that could cause confusion.
However, the rules do not consider a single mechanism in isolation on a section. All of the mechanisms on a section are considered when the rule is applied for that section.

Reply:
Copy from private discussion:
> Arbor and Neuron treat this fundamently diffent. In Neuron's model, Nernst can be used on a per-section basis in alternation with fixed reversal potentials. For Arbor, it is used globally, per ion species.

@halfflat on the motivation for not allowing it per region in Arbor,
- CVs could be partially covered by the Nernst and const methods for the same ion
- This cannot happen in Neuron, as each CV belongs to exactly one Section

Reply:
Let's add a comment about this final difference ARB ./. NRN ie

> In Arbor, `nernst` is applied globally at the level of ion species, while in Neuron different parts of the morphology may use different mechanisms for the revpot. This is the result of the difference in the models of discretisation (in Arbor `regions` may partially cover CVs).

and then I am happy :D